packages/backend/src/old/services/prices/PriceService.ts

```diff
@@ -63,4 +60,80 @@ export class PriceService {
 
     return new Map(entries)
   }
+
+  private async handleCache(
+    token: TokenInfo,
+    dates: SimpleDate[]
+  ): Promise<PriceHistoryPoint[]> {
+    const end = dates[dates.length - 1]
+
+    const prices: PriceHistoryPoint[] = []
+    let earliestUnknownDate = undefined
+
+    for (let i = 0; i < dates.length; i++) {
+      const cachedPrice = await this.getCachedPrice(token, dates[i])
+      if (!cachedPrice) {
+        earliestUnknownDate = dates[i]
+        this.logger.info('Cache miss', {
+          token: token.coingeckoId,
+          date: earliestUnknownDate.toString(),
+        })
+        break
+      } else {
+        prices.push(cachedPrice)
+      }
+    }
+
+    if (earliestUnknownDate !== undefined) {
+      const coingeckoPrices =
+        await this.coingeckoQueryService.getUsdPriceHistory(
+          CoingeckoId(token.coingeckoId),
+          new UnixTime(earliestUnknownDate.toUnixTimestamp()),
+          new UnixTime(end.toUnixTimestamp()),
+          'daily'
+        )
+      prices.push(...coingeckoPrices)
+
+      if (coingeckoPrices)
+        this.logger.info('Coingecko API call', {
+          token: token.coingeckoId,
+          amountOfPricesFetched: coingeckoPrices.length,
+        })
+
+      coingeckoPrices?.map((price) => {
```

Comment:
map implies that we care about the result. You could use `.forEach` but ideally just `for...of`